### PR TITLE
(feat) turn off auto docker volume mount if ARM64 as nested virtualisation isn't support on m1

### PR DESCRIPTION
**Pull request by eoinmcafee00**
Only change it for drone because right now delegate does not support arm64

### CODE REVIEW

File: engine/engine.go [hunk 1]
```diff
@@ -128,6 +130,10 @@ func (eng *Engine) Setup(ctx context.Context, specv runtime.Spec) error {
 		Files:     spec.Files,
 	}
 
+	if instance.Arch == ARM64 {
```
Comment (tphoney) [inline engine/engine.go:133]: can you put `const ArchARM64 = "arm64"` into oshelp similar to https://github.com/drone-runners/drone-runner-aws/pull/117/files#diff-2eb7297b270d1432e87ed95c084b3db7095e82eb638e3ae8e52e30af4f658774

**Review by tphoney — CHANGES_REQUESTED** [review]

File: engine/engine.go (hunk 1 above)
Comment (eoinmcafee00) [reply engine/engine.go]: done

**Review by tphoney — CHANGES_REQUESTED** [review]
this needs to be paired with os type or it will disable docker for ubuntu arm64

File: engine/engine.go [hunk 2]
```diff
@@ -128,6 +130,10 @@ func (eng *Engine) Setup(ctx context.Context, specv runtime.Spec) error {
 		Files:     spec.Files,
 	}
 
+	if instance.Arch == oshelp.ArchArm64 {
```
Comment (shubham149) [inline engine/engine.go:133]: Add a check that os needs to be darwin

File: engine/engine.go [hunk 3]
```diff
@@ -128,6 +130,10 @@ func (eng *Engine) Setup(ctx context.Context, specv runtime.Spec) error {
 		Files:     spec.Files,
 	}
 
+	if instance.Platform == oshelp.OSMac && instance.Arch == oshelp.ArchArm64 {
```
Comment (tphoney) [inline engine/engine.go:133]: can you put a comment in here, to explain what is happening. for future peeps

Comment (tphoney) [reply engine/engine.go:134]: you are explaining the code in the line below, you do not explain why.

```
// Currently the OSX m1 architecture does not enable nested virtualisation, so we disable docker.
```

**Pull request closed without merging**